pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1449,6 +1449,15 @@ private ByteBuf processMessageChunk(ByteBuf compressedPayload, MessageMetadata m
         // discard message if chunk is out-of-order
         if (chunkedMsgCtx == null || chunkedMsgCtx.chunkedMsgBuffer == null
                 || msgMetadata.getChunkId() != (chunkedMsgCtx.lastChunkedMessageId + 1)) {
+            // Filter duplicated chunks instead of discard it.
+            if (chunkedMsgCtx == null || msgMetadata.getChunkId() <= chunkedMsgCtx.lastChunkedMessageId) {
```

Comment:
why `chunkedMsgCtx == null` here?

Reply:
It happens when there is no cached `chunkedMsgCtx` while the consumer receives a chunk whose id is not 0. Normally, the consumer should not receive the chunk-i (i > 0) before receiving chunk-0.

Reply:
It happens when the first chunk is duplicated. 
```
Chunk-0
Chunk-0
Chunk-1
Chunk-2
```